### PR TITLE
arm: beetle/mps2: device tree and debug support

### PR DESCRIPTION
**Pull request by fvincenzo**
Extend DTS support for beetle and mps2 and add debug features for beetle via OpenOCD.

### CODE REVIEW

File: arch/arm/soc/arm/mps2/soc_devices.h [hunk 1]
```diff
@@ -10,6 +10,7 @@
 #include <soc_memory_map.h>
 #include <soc_irq.h>
 
+#if !defined(CONFIG_HAS_DTS)
```
Comment (galak) [inline arch/arm/soc/arm/mps2/soc_devices.h]: is there any point to support !CONFIG_HAS_DTS?

File: arch/arm/soc/arm/beetle/soc.h [hunk 2]
```diff
@@ -113,9 +113,11 @@
 /* CMSDK APB Dual Timer */
 #define CMSDK_APB_DTIMER _BEETLE_DTIMER_BASE
 
+#ifndef CONFIG_HAS_DTS
```
Comment (galak) [inline arch/arm/soc/arm/beetle/soc.h]: any reason we can't just remove ifndef CONFIG_HAS_DTS case?

File: dts/arm/v2m_beetle.dts [hunk 3]
```diff
@@ -43,6 +43,11 @@
 			interrupts = <IRQ_UART1 3>;
 			current-speed = <115200>;
 		};
+
+		wdog0: wdog@40008000 {
+			compatible = "arm,cmsdk-wdog";
```
Comment (galak) [inline dts/arm/v2m_beetle.dts]: Make the compatible either arm,cmsdk-watchdog or arm,cmsdk-wdt

File: dts/arm/yaml/arm,cmsdk-wdog.yaml [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+
+description: >
+    This binding gives a base representation of the ARM CMSDK WATCHDOG
+
+properties:
+  - compatible:
+      type: string
+      category: required
+      description: compatible strings
+      constraint: "arm,cmsdk-wdog"
```
Comment (galak) [inline dts/arm/yaml/arm,cmsdk-wdog.yaml]: Make the compatible either arm,cmsdk-watchdog or arm,cmsdk-wdt

File: dts/arm/yaml/arm,cmsdk-wdog.yaml [hunk 5]
```diff
@@ -0,0 +1,21 @@
+---
+title: ARM CMSDK WATCHDOG
+id: arm,cmsdk-wdog
```
Comment (galak) [inline dts/arm/yaml/arm,cmsdk-wdog.yaml]: Make the compatible either arm,cmsdk-watchdog or arm,cmsdk-wdt

**Review by galak — CHANGES_REQUESTED** [review]
In general if we are getting things from DT, can we remove the code related to !CONFIG_HAS_DTS?

File: arch/arm/soc/arm/mps2/soc_devices.h (hunk 1 above)
Comment (fvincenzo) [inline arch/arm/soc/arm/mps2/soc_devices.h]: I was thinking to keep both the options, but if the indication is device tree only, I can remove !CONFIG_HAS_DTS bits.

Comment (galak) [inline arch/arm/soc/arm/mps2/soc_devices.h]: Yeah, we are only going to support CONFIG_HAS_DTS on arm

**Review by galak — APPROVED** [review]

**Pull request merged**